app/javascript/lib/service-worker-cache.js

```diff
@@ -0,0 +1,55 @@
+import { getSCPContext } from '~/providers/SCPContextProvider'
+
+const scpContext = getSCPContext()
+const env = scpContext.environment
+const version = scpContext.version
+
+/** Whether to fetch data from service worker cache */
+export const isServiceWorkerCacheEnabled = scpContext.isServiceWorkerCacheEnabled
+
+// Cache keys are like a mix of database server and collection/table names:
+// they identify a cache store for individual cache entries.  Here we set up
+// cache keys specific to each SCP environment and version.
+const serviceWorkerCacheKeyStem = `scp-${env}`
+const serviceWorkerCacheKey = `${serviceWorkerCacheKeyStem}-${version}`
+
+/**
+ * Fetch, leveraging service worker cache if enabled and available
+ *
+ * TODO (SCP-4508): Account for same URL, different sign-in state in service worker cache
+ */
+export async function fetchServiceWorkerCache(url, init) {
+  const swCache = await caches.open(serviceWorkerCacheKey)
+  let response = await swCache.match(url)
+  let hitOrMiss = 'hit'
+  if (typeof response === 'undefined') {
+    response = await fetch(url, init).catch(error => error)
+    await swCache.put(url, response.clone())
+    hitOrMiss = 'miss'
+  }
+  console.debug(`Service worker cache ${hitOrMiss} for SCP API fetch of URL: ${url}`)
```

Comment:
```suggestion
  console.debug(`Service worker cache ${isHit ? 'hit' : 'miss'} for SCP API fetch of URL: ${url}`)
```

Reply:
Done as part of work per my [other comment](https://github.com/broadinstitute/single_cell_portal_core/pull/1565#discussion_r923602407).